firewood/src/merkle.rs

```diff
@@ -1472,6 +1931,37 @@ pub fn from_nibbles(nibbles: &[u8]) -> impl Iterator<Item = u8> + '_ {
     nibbles.chunks_exact(2).map(|p| (p[0] << 4) | p[1])
 }
 
+#[derive(Debug)]
+struct Overlap<'a, T> {
```

Comment:
I think here deserves a comment to indicate where to check the overlap part. If I understand correctly based on `from`, we are only checking from the start and don't care about overlap if in the middle or end.

Reply:
Good call out. I'll change the name.